go/arrow/array/data.go

```diff
@@ -190,9 +190,32 @@ func (d *Data) SetDictionary(dict arrow.ArrayData) {
 	}
 }
 
+// Returns the size of Data in bytes
+func (d *Data) SizeInBytes() uint64 {
```

Comment:
The convention for godoc is that the comment should start with the name of the method, i.e. `// SizeInBytes returns ....`

I think we should also include the fact that this does the recursion into the children in the comment so that it's obvious in the documentation and users don't have to read the code to find that out.

Reply:
Good call, changed.